.github/workflows/dependency-update.yaml

```
@@ -0,0 +1,58 @@
+name: Update Matter SDK submodule
+
+on:
+  schedule:
+    # Runs at 04:30 UTC every day
+    - cron: '30 04 * * *'
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-connectedhomeip-submodule:
+    runs-on: ubuntu-latest
+    name: Update Matter SDK submodule
+
+    steps:
+      # Checkout the repository
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          submodules: false
+
+      # Update submodule (not recursive)
+      - name: Update Submodule
+        id: update-submodule
+        run: |
+          git submodule update --init connectedhomeip/
+          short_hash_before=$(cd connectedhomeip && git rev-parse --short HEAD)
+          echo "short_hash_before=${short_hash_before}" >> $GITHUB_OUTPUT
+          git submodule update --remote connectedhomeip/
+          git add connectedhomeip/
+          if git diff-index --quiet HEAD; then
+              echo "empty=true" >> $GITHUB_OUTPUT
+              exit 0
+          fi
+          short_hash=$(cd connectedhomeip && git rev-parse --short HEAD)
+          echo "short_hash=${short_hash}" >> $GITHUB_OUTPUT
+
+      # Create Pull Request only if there were changes
+      - name: Create Pull Request
+        if: ${{ steps.update-submodule.outputs.empty != true }}
```

Not sure it will work without quotes, so:
```suggestion
        if: ${{ steps.update-submodule.outputs.empty != "true" }}
```

---

As per my testing, it does. But this is more explicit.